src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -329,9 +365,6 @@ public void write() {
                 break;
               }
             } else {
-              SinkRecord projectedRecord = compatibility.project(record, null, currentSchema);
```

Comment:
This is an important reordering that is required in order to be able to rotate based on a record's timestamp.